CourseProject/intervals/intervals.js

```diff
@@ -1,26 +1,105 @@
+        this.#intervalResults = new IntervalResults();
 
-        this.#date1.addEventListener("change", this.#conditionOfChoosingDate2)
+        this.#dateInput1.addEventListener("change", this.#conditionOfChoosingDate2);
+
+        this.#week.addEventListener("click", this.#weekPreset);
+        this.#month.addEventListener("click", this.#monthPreset);
+
+        this.#calcResult.addEventListener("click", this.#calculate);
     }
 
     #conditionOfChoosingDate2 = () => {
-        console.log(this.#date1.value)
-        if (this.#date1.value === "") {
-            this.#date2.setAttribute("disabled", "");
+        if (this.#dateInput1.value === "") {
+            this.#dateInput2.setAttribute("disabled", "");
         } else {
-            this.#date2.removeAttribute("disabled");
+            this.#dateInput2.removeAttribute("disabled");
         }
-        this.#date2.setAttribute("min", this.#date1.value)
-    }
-}
+        this.#dateInput2.setAttribute("min", this.#dateInput1.value);
+    };
+
+    #weekPreset = () => {
+        let date1;
+        if (this.#dateInput1.value === "") {
+            date1 = new Date();
+            this.#dateInput1.value = this.#datesHelper.format(date1);
+        } else {
+            date1 = new Date(this.#dateInput1.value);
+        }
+        const date2 = this.#datesHelper.addWeek(date1)
+       
+        this.#dateInput2.value = this.#datesHelper.format(date2);
+        this.#dateInput2.removeAttribute("disabled");
+    };
+    #monthPreset = () => {
+        let date1;
+        if (this.#dateInput1.value === "") {
+            date1 = new Date();
+            this.#dateInput1.value = this.#datesHelper.format(date1);
+        } else {
+            date1 = new Date(this.#dateInput1.value);
+        }
+        const date2 = this.#datesHelper.addMonth(date1)
+       
+        this.#dateInput2.value = this.#datesHelper.format(date2);
+        this.#dateInput2.removeAttribute("disabled");
+    };
+    #calculate = () => {
 
+        const firstDate = new Date(this.#dateInput1.value);
+        const secondDate = new Date(this.#dateInput2.value);
 
+        let numberOfDays;
+        if (this.#selectDays.value === "alldays") {
+            numberOfDays = this.#datesHelper.getAlldaysCount(firstDate, secondDate);
+        } else if (this.#selectDays.value === "weekdays") {
+            numberOfDays = this.#datesHelper.getweekDayCount(firstDate, secondDate);
+        } else {
+            numberOfDays = this.#datesHelper.getweekEndCount(firstDate, secondDate);
+        }
 
+        let output;
+        if (this.#selectIntervals.value === "days") {
+            output = numberOfDays;
+        } else if (this.#selectIntervals.value === "hours") {
+            output = numberOfDays*24;
+        } else if (this.#selectIntervals.value === "minutes") {
+            output = numberOfDays*24*60;
+        } else {
+            output = numberOfDays*24*60*60;
+        }
```

Comment:
```suggestion
switch (this.#selectIntervals.value) {
    case "days":
        output = numberOfDays;
        break;
    case "hours":
        output = numberOfDays*24;
        break;
    case "minutes":
        output = numberOfDays*24*60;
        break;
    default:
        output = numberOfDays*24*60*60;
}
```
в таких випадках switch/case буде зручніше та швидше, за рахунок внутрішніх оптимізацій, ніж if/else